### PR TITLE
allow the user to submit a challenge

### PR DESCRIPTION
**Pull request by alainburindi**
#### Description of changes made
This PR will allow the user to submit a new challenge

#### Type of change

- [x] New feature (non-breaking changes which add the new functionalities)
- [ ] Bug fix (non-breaking changes which fix an issue)
- [ ] Documentation (added or improved the documentation)
- [ ] Breaking change (fix or a feature that would cause existing feature not to work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?

- After cloning the [repo](https://github.com/kisanola/AlphaCodeFrontend.git)
- Check out this branch by `ft-submit-challenge`
- Run `yarn or npm install` to install the project dependencies
- Run `npm t`

- [ ] Unit
- [x] Integration
- [ ] End to end

#### How should this be manually tested?

> **Provide the steps to manually test the functionalities implemented in this PR**

#### Add the link to the to Trello Board?
[Users should be able to submit a new challenge](https://app.gitkraken.com/glo/card/5499382e2bad4638b669983b3ec099c6)
#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have ensured coding conventions for my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests which prove the effectiveness of my work
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

**Comment by oesukam** [comment]
@alainburindi Just to clarify. This task is for creating/submitting a new chanllenge instead of submitting a solution

**Comment by espoirMur** [comment]
@alainburindi, thanks for this PR!
Don't forget to describe your PR according to the PR description template we have...

**Comment by alainburindi** [comment]
> @alainburindi Just to clarify. This task is for creating/submitting a new chanllenge instead of submitting a solution

Hi @oesukam, Sorry for the delayed reply. it seems like I didn't understand the task very well. thanks for your clarification

**Comment by alainburindi** [comment]
> @alainburindi, thanks for this PR!
> Don't forget to describe your PR according to the PR description template we have...

Sure I will

File: src/models/index.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ const dbURLS: DBInterface = {
   development: DATABASE_URL,
 };
 
-const URL: string = dbURLS[NODE_ENV] as string;
+const URL: string = dbURLS['development'] as string;
```
Comment (espoirMur) [inline src/models/index.ts]: Why are you changing this @alainburindi ?
Read the app environment form the command line!
This is what is making your tests failing when running on CI!
On ci, we are in `test` environment

**Review by espoirMur — CHANGES_REQUESTED** [review]

**Comment by alainburindi** [comment]
I changed it last week and I forgot to change it back. I think that's the
problem

File: src/helpers/constants/statuses.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default {
+  created: 201,
```
Comment (espoirMur) [inline src/helpers/constants/statuses.ts]: @alainburindi, this is awesome, can you please add the remaining status codes?

File: src/models/index.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import mongoose from 'mongoose';
 import 'dotenv/config';
 
-const { NODE_ENV, DATABASE_URL, DATABASE_URL_TEST, DATABASE_URL_PRODUCTION } = process.env;
+const { NODE_ENV = '', DATABASE_URL, DATABASE_URL_TEST, DATABASE_URL_PRODUCTION } = process.env;
```
Comment (espoirMur) [inline src/models/index.ts:4]: I am not getting what the ='' is doing here..
? can you please clarifiry? @alainburindi

Comment (alainburindi) [reply src/models/index.ts:4]: I'm having issues with environment variables, they are undefined at runtime

Comment (espoirMur) [reply src/models/index.ts:4]: Rebase from develop and check , we are setting them on runtime

Comment (espoirMur) [reply src/models/index.ts:4]: Screen Shot 2020-05-02 at 18 38 47
Check this.....

Comment (espoirMur) [reply src/models/index.ts:4]: The way we are setting them up...

Comment (alainburindi) [reply src/models/index.ts:4]: I have the same setup. the problem I was having was on runtime. TypeScript couldn't compile, it was saying expecting string but undefined was given


scri

Comment (espoirMur) [reply src/models/index.ts:4]: Are you sure we you have a `.env` file in your project folder?
What did you get when you do console.log(process.env)?

Comment (espoirMur) [reply src/models/index.ts:4]: @alainburindi !!!

Comment (oesukam) [reply src/models/index.ts:4]: @alainburindi You can update the global interface

File: src/middlewares/checkAuth.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ const checkAuth = async (req: any, res: Response, next: NextFunction): Promise<a
   }
 
   jwt.verify(
-    token,
+    authorization,
```
Comment (oesukam) [inline src/middlewares/checkAuth.ts:22]: Does the token from the header not being prefixed such as with `Bearer` or `JWT`?

**Pull request merged**